### PR TITLE
Update Flow. Remove Bluebird

### LINKED ISSUE #2725: Remove bluebird dependency
With our new supported browser policy we can now remove bluebird as a dependency. We already removed it from the native desktop client part and we should change the flow typing.

---

Test notes:
 - [ ] Check that the app generally works on all supported browsers (we removed async-to-generator translation)
 - [ ] Check that signup still works
 - [ ] Check that creating a session works (password login)
 - [ ] Check that external login still works
 - [ ] Check that download all in desktop works
 - [ ] Check that download all with duplicate filenames works in desktop and names are changed
 - [ ] Check that search index generally works
 - [ ] Check that search works on iOS
 - [ ] Check that updating event does not show error if the event is already deleted
 - [ ] Check that while creating event alarms are loaded one by one
 - [ ] Check that calendar event invites generally still work
 - [ ] Check that attaching and uploading files works and is done one by one
 - [ ] Check that loading missed events after reconnect works
 - [ ] Adding/removing calendars works
 - [ ] Contact viewing still works
 - [ ] Async dropdowns
 - [ ] Showing progreess in list (initially and later)
 - [ ] Swipe actions in the list
 - [ ] Date picker
 - [ ] Contact form requests
 - [ ] Recover login error handling (like wrong password)
 - [ ] Takeover error handling (like wrong password)
 - [ ] Login view: invalid gift card handling
 - [ ] Purchasing gift card
 - [ ] Redeeming gift card
 - [ ] Reach size limit in mail editor
 - [ ] Mail export w/ attachments
 - [ ] Inbox rules are still applied (both when loading the least and from WS updates)
 - [ ] Moving mails
 - [ ] Deleting mails
 - [ ] Viewing/scaling of mails still works, loading external content still works
 - [ ] Scrolling mail in viewer with shortcuts
 - [ ] Approving 2FA
 - [ ] U2F register/login
 - [ ] Moving multiple mails from different mailboxes is not possible in search
 - [ ] Adding alias and running into limit
 - [ ] Changing mail signature
 - [ ] Configuring out of office
 - [ ] Displaying of template/knowledgebase groups in settings
 - [ ] Deleting account
 - [ ] Payment data dialog shows
 - [ ] Paypal dialog from payment viewer works (add paypal)
 - [ ] Change storage capacity
 - [ ] Switch subscription
 - [ ] Templates load and can be inserted

### PR DESCRIPTION
**Pull request by charlag**
Close #2725 #3205
Depends on oxmsg branch, see https://github.com/tutao/oxmsg/pull/2/

Code generator & admin updated in flow-bluebird branch.

### CODE REVIEW

**Comment by johnbotris** [comment]
OXMSG changes have been published, so we can change the dependency to `"oxmsg": "0.0.7"`

File: make.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ if (opts.watch) {
 
 const flowPromise = new Promise((resolve, reject) => {
 	// It's better to set listener right away
-	spawn(flow, ["--quiet"], {stdio: "inherit"}).on("exit", resolve).on("error", reject)
```
Comment (johnbotris) [inline make.js:36]: i like it better when it's quiet

File: package.json [hunk 2]
```diff
@@ -16,14 +16,13 @@
...
 	"dependencies": {
-		"bluebird": "3.7.2",
 		"dompurify": "2.2.7",
 		"electron-builder": "22.11.4",
 		"keytar": "7.2.0",
 		"linkifyjs": "^3.0.0-beta.3",
 		"luxon": "1.26.0",
 		"mithril": "2.0.4",
-		"@tutao/oxmsg": "0.0.6",
+		"@tutao/oxmsg": "https://github.com/tutao/oxmsg.git#6e704f571b2307bb91e9af1020624a02b85b3f33",
```
Comment (johnbotris) [inline package.json:25]: `"@tutao/oxmsg": "0.0.7"`

**Review by johnbotris — CHANGES_REQUESTED** [review]
Looks good besides the things pointed out.
Some thoughts:
* It could be good to have some kind of wrapper that gives us `Promise.isPending()`/`Promise.isFulfilled()` type functionality since we use that pattern in a few places still
* We have `PromiseUtils.tap` for some of the `Promise.tap` replacements (lots of `.then(it => ...; return it)`)
* an `ofClass` that accepts multiple classes could be useful
* Replacements of `Promise.each` with `promiseMap` could probably be better off as for-each loops (although we could introduce concurrency more easily if we leave it?)
* lots of `return promiseMap(...).then(noOp)` which could also just be for-each  loops, or just `await` and drop the `then`

(I recognize the last two were probably from find/replace and it's probably more effort than it's worth to fix)

File: src/desktop/DesktopDownloadManager.js [hunk 3]
```diff
@@ -44,7 +43,7 @@ export class DesktopDownloadManager {
 		       .on('will-download', (ev, item) => this._handleDownloadItem(item))
 	}
 
-	async downloadNative(sourceUrl: string, fileName: string, headers: {v: string, accessToken: string}): Promise<{statusCode: string, statusMessage: string, encryptedFileUri: string}> {
+	async downloadNative(sourceUrl: string, fileName: string, headers: {|v: string, accessToken: string|}): Promise<{statusCode: string, statusMessage: string, encryptedFileUri: string}> {
```
[the remark this answers is not in the document]
Comment (vitoreiji) [reply src/desktop/DesktopDownloadManager.js:46]: I guess explicit exact or inexact notation is still recommended.
- https://medium.com/flow-type/on-the-roadmap-exact-objects-by-default-16b72933c5cf
- https://github.com/facebook/flow/releases/tag/v0.112.0

**Comment by charlag** [comment]
>     * It could be good to have some kind of wrapper that gives us `Promise.isPending()`/`Promise.isFulfilled()` type functionality since we use that pattern in a few places still

We considered that but we could get away without it. I actually think that a proper state machine is better than checking things on promises in most cases.

>     * We have `PromiseUtils.tap` for some of the `Promise.tap` replacements (lots of `.then(it => ...; return it)`)

We have but I'm not sure if it's less wordy or not. Maybe it's less error-prone at least

>     * an `ofClass` that accepts multiple classes could be useful

I think we are already abusing exceptions for business logic and I would love us to do it less. If you need multiple exceptions just use `instanceof` manually.

>     * Replacements of `Promise.each` with `promiseMap` could probably be better off as for-each loops (although we could introduce concurrency more easily if we leave it?)

I'm not sure I understand your point. You propose to not use `promiseMap` and just write for loop each time?

**Review by johnbotris — APPROVED** [review]

**Pull request merged**